aten/src/TH/generic/THTensorMath.c

```diff
@@ -3092,6 +3092,81 @@ void THTensor_(atan2)(THTensor *r_, THTensor *tx, THTensor *ty)
   TH_TENSOR_APPLY3(real, r_, real, tx, real, ty, *r__data = TH_MATH_NAME(atan2)(*tx_data,*ty_data););
 }
 
+// Some helpers for THTensor_(digamma)
+accreal THTensor_(polevl)(accreal x, accreal *A, size_t len);
+accreal THTensor_(digamma_asym)(accreal x);
+real THTensor_(digamma_one)(real x);
+
+/*
+ * Evaluates A[0] x^len + A[1] X^{len-1} + ... + A[len-1]
+ */
+inline accreal THTensor_(polevl)(accreal x, accreal *A, size_t len) {
+  accreal result = 0;
+  for (size_t i = 0; i <= len; i++) {
+    result = result * x + A[i];
+  }
+  return result;
+}
+
+inline accreal THTensor_(digamma_asym)(accreal x) {
+  static accreal A[] = {
+     8.33333333333333333333E-2,
+    -2.10927960927960927961E-2,
+     7.57575757575757575758E-3,
+    -4.16666666666666666667E-3,
+     3.96825396825396825397E-3,
+    -8.33333333333333333333E-3,
+     8.33333333333333333333E-2
+  };
+
+  accreal y = 0;
+  if (x < 1.0e17) {
+	  accreal z = 1.0 / (x * x);
+	  y = z * THTensor_(polevl)(z, A, 6);
+  }
+  return log(x) - (0.5 / x) - y;
+}
+
+/*
+ * Algorithm adapted from Cephes
```

Comment:
Does adapted == pretty much copied, or did you rewrite it based on Cephes? I'm not sure what's the license, but we might need to include it here.

Reply:
I think the answer to that is pretty much copied: the algorithm is the same, except for one case (integers from 1 - 10). Here's the original cephes code I used for reference: https://github.com/jeremybarnes/cephes/blob/master/misc/psi.c -- I can include the license there here.

Reply:
Cephes doesn't state its license clearly except from saying that it "can be used freely". I think it's best to ask the author, just to be sure. Aparently [that's what authors of torch-cephes did](https://github.com/deepmind/torch-cephes/blob/master/LICENSE.txt).